table/block_based/block.cc

```diff
@@ -659,17 +648,66 @@ void IndexBlockIter::DecodeCurrentValue(uint32_t shared) {
   }
 }
 
-// Binary search in restart array to find the first restart point that
-// is either the last restart point with a key less than target,
-// which means the key of next restart point is larger than target, or
-// the first restart point with a key = target
+template <class TValue>
+void BlockIter<TValue>::ScanAfterBinarySeek(const Slice& target, uint32_t index,
+                                            bool is_index_key_result,
+                                            const Comparator* comp) {
+  // Linear search (within restart block) for first key >= target
+  SeekToRestartPoint(index);
+  Next();
```

Comment:
A comment here might help. It took me a little while to figure out that this ```Next()``` actually doesn't advance to the next key, but just ends up decoding the key at the restart point we seek'd to.